### DISCUSSION TITLE
Library Installation problem

### DISCUSSION THREAD

**Discussion by samermashaan** [stephenberry/glaze#940] [General]
Hi,

I'm trying to install and use the library on a C++ project (Windows using VSCode with G++ compiler) and included the gaze directory in the following path: ${workspaceFolder}/include/glaze

I'm trying to import the library on my main.cpp which is located on the path
${workspaceFolder}/lib/lib.cpp

I've tried to include the path but still getting path errors in every header file

Can anyone suggest an installation guide link or explain if I'm including the library in the wrong way?

Regards

**Comment by stephenberry** [comment]
`${workspaceFolder}/include/gaze` spells `gaze` rather than `glaze`, is this the problem?

**Reply by samermashaan** [reply]
Sorry i had a mistake in forming the post. I corrected it.
It's is imported however with the correct naming, and still the glaze.hpp header cant find other included header files

**Reply by stephenberry** [reply]
I would need some more configuration information to help. Are you using CMake? If you use FetchContent_Declare in your CMakeLists.txt like the README suggests, then you should just be able to add `target_link_libraries(${PROJECT_NAME} PRIVATE glaze::glaze)` to your CMake and get the proper includes. If you want to refer to them directly then you'll find them in the `build/_deps/...` directory in VSCode.

**Reply by samermashaan** [reply]
Yes I'm using the example found here https://github.com/stephenberry/glaze_example/tree/main
I used the following cmake command to generate the build `cmake -S . -B build`.

I tried like the example using in example.cpp
#include "glaze/glaze.hpp"
#include <glaze/glaze.hpp>
#include "../build/_deps/glaze-src/include/glaze/glaze.hpp"

The last one from your suggestion did find `glaze.hpp` file, but now I'm getting another error referer to  `glaze/binary.hpp` which included in `glaze.hpp` can't be found.

**Reply by stephenberry** [reply]
Yeah, `#include "../build/_deps/glaze-src/include/glaze/glaze.hpp"` won't work because you need to have the path to glaze's include directory in your include paths, since the rest of the includes need to be `"glaze/..."`, which is a standard practice in C++ development.

I think you have a basic setup problem in VSCode, but I'm not sure what it is. I can run `cmake -S . -B build` in VSCode on the `glaze_example` project and everything builds and runs out of the box.

**Reply by samermashaan** [reply]
Ok good to know, ill try then another IDE.
Thanks for your support :)
